README_EN.md

```diff
@@ -0,0 +1,240 @@
+
+```Swift
+  let intent = FRDIntent(clazz: SecondViewController.self)
+  let manager = FRDControllerManager.sharedInstance
+  manager.startController(source: self, intent: intent)
+```
+
+#### Launch view controller by url
+
+```Swift
+  let intent = FRDIntent(uri: URL(string: "/frodo/firstview")!)
+  let manager = FRDControllerManager.sharedInstance
+  manager.startController(source: self, intent: intent)
+```
+
+#### Launch view controller to get result data
+
+```Swift
+  let intent = FRDIntent(clazz: ThirdViewController.self)
+  intent.putExtra(name: "text", data: "Text From Source")
+  let manager = FRDControllerManager.sharedInstance
+  manager.startControllerForResult(source: self, intent: intent, requestCode: RequestText)
+```
+
+Someone, maybe the caller itself must conform the `FRDIntentForResultSendable` protocol to get the feedback result data from the recipient.
+
+```Swift
+  extension ViewController: FRDIntentForResultSendable {
+
+    func onControllerResult(requestCode: Int, resultCode: FRDResultCode, data: Intent) {
+      if (requestCode == RequestText) {
+        if (resultCode == .ok) {
+          let text = data.extra["text"]
+          print("Successful confirm get from destination : \(text)")
+        } else if (resultCode == .canceled) {
+          let text = data.extra["text"]
+          print("Canceled get from destination : \(text)")
+        }
+      }
+    }
+
+  }
+```
+
+The recipient must conform the `FRDIntentForResultReceivable` protocol to receive the Intent. `FRDIntentForResultReceivable` is `FRDIntentReceivable`'s sub-protocol, and has two more variables.
+
+```Swift
+  var data: [String: Any]?
+  var requestCode: Int?
+```
+
+#### Custom transition animation
+
+In FRDIntent, the transition is abstracted as `FRDControllerDisplay` protocol. There are already two implementations: `FRDPushDisplay` and `FRDPresentationDisplay` of `FRDControllerDisplay`.
+
+Your custom transition should conform this protocol `FRDControllerDisplay`. When the caller want to launch a view controller, set the `FRDIntent`'s `controllerDisplay` variable to the object presenting your custom tranistion.
+
+FRDIntent's default transition is `FRDPushDisplay` within `starController`, or `FRDPresentationDisplay` within `startControllerForResult`.
+
+## URLRoutes
+
+FRDIntent/URLRoutes is an URL Router. FRDIntent/URLRoutes sustains a map between url and block. Any external app can send the url request to execute the corresponding block.
+
+iOS provides a solution to call an app from another app basing url request. FRDIntent/URLRoutes brings the ease of usage of this solution. From this point, FRDIntent/URLRoutes is similiar to many URL Routers in community like [JLRoutes](https://github.com/joeldev/JLRoutes).
+
+The implementation of URLRoutes in FRDIntent benefits the cooperation between FRDIntent/Intent and FRDIntent/URLRoutes. They can provide together a complete solution to handle internal call and external call of view controller.
```

Comment:
In a sense, FRDIntent/URLRoutes is similar to many URL Routers in the community, [JLRoutes](https://github.com/joeldev/JLRoutes) for instance. But writing routes with FRDIntent/URLRoutes allows you to organically integrate the routes with FRDIntent/Intent, which makes it a lot simpler to navigate through internal calls and external calls.